tests/InputFiles/GeneralizedHarmonic/KerrSchild.yaml

```diff
@@ -46,11 +46,13 @@ EventsAndTriggers:
   ? EveryNSlabs:
       N: 2
       Offset: 0
-  : - ObserveErrorNorms
+  : - ObserveErrorNorms:
+        SubfileName: Errors
```

Comment:
Do you have a use case for changing the subfile name for observations in the input file? I wonder if it would be more user-friendly to have the subfile name be determined at compile-time, so it is the same for every run of an executable. Code that reads data from a particular executable's output can then rely on data being present at predictable locations in the subfile. What do you think?

Reply:
That doesn't work. If you want to be able to choose how often you want to observe and what to observe in the input file, you have to specify where you write it. Basically, this is the sole reason any of the changes actually allow you to have multiple reduction and volume observations running.

Reply:
Right of course 🤦‍♂️

Reply:
Wait I can't think of a use case now where you would have the same compile-time event observe multiple times, i.e. need to set the subfile name at runtime.. what would be an example?

Reply:
You want to dump the lapse every 10 time steps as volume data and the spacetime metric every 100. You want to dump the errors every 3 time steps and the constraints every 12. Literally anything where you want to have more than one event that does a reduction or observation, even if they are of different fields. This is what Tom tried to do: https://github.com/sxs-collaboration/spectre/issues/2363 

There are basically endless examples of where you'd want to observe different things at different frequencies, and requiring that be set at compile time would make the code horribly painful to use.

Reply:
What confuses me is that I thought the observations could be written into the same subfile even when they happen at different frequencies. If I understand you correctly, you would write the lapse into one subfile every 10 timesteps and the metric in another subfile every 100 timesteps. Can't they both be written in the same subfile, that always has the same name?

Reply:
Regarding #2363: To resolve that issue the subfiles don't need to be set at runtime. The `ObserveErrorNorms` writes into `error_norms.dat` and `ObserveVolumeIntegrals` into `volume_integrals.dat`, for example.

Reply:
No, they can't be because at every 100 time step you are getting data from 2 actions going into the same file. You now need to keep track of when you get data for only the lapse and when for both the lapse and the metric. This is going to make the HDF5 file format much more complicated than it currently is, and I don't really see any benefit to trying to do something that complicated. It's been the plan to use different subfile names since when we first discussed the observers like 4 years ago.

Sure, but is there what happens if you want to observe the error norms of the metric every 10 steps and the metric and auxiliary variables every 100 steps? This is not at all supported by the dat files and is basically impossible even at the HDF5 level without making the file format a complete nightmare. I guess you could write `0.0` at the times you don't observe it, but I expect people to be more confused that the error jobs to zero than that they are in different subfiles.

Reply:
Ok thanks for explaining 👍

Reply:
You're welcome! :)